ratis-client/src/main/java/org/apache/ratis/client/api/DataStreamOutput.java

```diff
@@ -28,7 +28,7 @@
 import java.util.concurrent.CompletableFuture;
 
 /** An asynchronous output stream supporting zero buffer copying. */
-public interface DataStreamOutput extends CloseAsync<DataStreamReply> {
```

Comment:
I think it is convenient to keep extending CloseAsync<DataStreamReply> so that the API is easier to be used.  We may provide a default closeAsync() implementation.

Reply:
@szetszwo Sorry, what should we do in closeAsync() ? Do we need to send rpc call to primary ?  If so, user maybe confused between writeAndCloseAsync and closeAsync. I delete `extends CloseAsync` so that user can only close by writeAndCloseAsync.

Reply:
If it is already closed, do nothing.  Otherwise, call writeAsync(DataStreamPacketByteBuffer.EMPTY_BYTE_BUFFER, WriteOption.CLOSE).

Reply:
>  If so, user maybe confused between writeAndCloseAsync and closeAsync. ...

Since writeAndCloseAsync is uncommon, I think more users will be confused with it.  :)

Reply:
@szetszwo  I see, thanks.